bundle/edu.gemini.itc/src/main/scala/edu/gemini/itc/service/ItcServiceImpl.scala

```diff
@@ -115,6 +134,10 @@ class ItcServiceImpl extends ItcService {
     ItcResult.forResult(s)
   }
 
-
+  private def spectroscopyResultCharts(recipe: SpectroscopyArrayRecipe): Result = {
+    val r = recipe.calculateSpectroscopy()
+    val s = recipe.serviceResult(r, false)
+    ItcResult.forResult(s)
```

Comment:
What do `r` and `s` consist of?  This must be something that already existed?

Reply:
These methods calculate the graph data via an ITC recipe. This is a copy of the existing `spoctroscopyResults` which in hindsight I could probably just reuse setting the headless value to false.
Originally I expected I needed to do some changes to the service result but in practice itc will always return the full data including graphs and calculated SN